### ISSUE TITLE
Seperate up the links/blocks properties in StructuredTextGraphQlResponse

### ISSUE THREAD

**Issue by BasixKOR** [datocms/react-datocms#50]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `react-datocms@3.0.6` for the project I'm working on.

This issue is a follow-up from [this forum post](https://community.datocms.com/t/bug-react-datocms-structuredtexts-incorrect-typings-on-blocks-links/2693).

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/react-datocms/dist/types/StructuredText/index.d.ts b/node_modules/react-datocms/dist/types/StructuredText/index.d.ts
index 5e55966..1ea35fd 100644
--- a/node_modules/react-datocms/dist/types/StructuredText/index.d.ts
+++ b/node_modules/react-datocms/dist/types/StructuredText/index.d.ts
@@ -25,19 +25,19 @@ declare type RenderRecordLinkContext<R extends StructuredTextGraphQlResponseReco
 declare type RenderBlockContext<R extends StructuredTextGraphQlResponseRecord> = {
     record: R;
 };
-export declare type StructuredTextPropTypes<R extends StructuredTextGraphQlResponseRecord> = {
+export declare type StructuredTextPropTypes<R1 extends StructuredTextGraphQlResponseRecord, R2 extends StructuredTextGraphQlResponseRecord> = {
     /** The actual field value you get from DatoCMS **/
-    data: StructuredTextGraphQlResponse<R> | StructuredTextDocument | Node | null | undefined;
+    data: StructuredTextGraphQlResponse<R1, R2> | StructuredTextDocument | Node | null | undefined;
     /** A set of additional rules to convert nodes to JSX **/
     customNodeRules?: RenderRule<H, T, F>[];
     /** A set of additional rules to convert marks to JSX **/
     customMarkRules?: RenderMarkRule<H, T, F>[];
     /** Fuction that converts an 'inlineItem' node into React **/
-    renderInlineRecord?: (context: RenderInlineRecordContext<R>) => ReactElement | null;
+    renderInlineRecord?: (context: RenderInlineRecordContext<R2>) => ReactElement | null;
     /** Fuction that converts an 'itemLink' node into React **/
-    renderLinkToRecord?: (context: RenderRecordLinkContext<R>) => ReactElement | null;
+    renderLinkToRecord?: (context: RenderRecordLinkContext<R2>) => ReactElement | null;
     /** Fuction that converts a 'block' node into React **/
-    renderBlock?: (context: RenderBlockContext<R>) => ReactElement | null;
+    renderBlock?: (context: RenderBlockContext<R1>) => ReactElement | null;
     /** Function that converts 'link' and 'itemLink' `meta` into HTML props */
     metaTransformer?: TransformMetaFn;
     /** Fuction that converts a simple string text into React **/
@@ -49,4 +49,4 @@ export declare type StructuredTextPropTypes<R extends StructuredTextGraphQlRespo
     /** @deprecated use customNodeRules **/
     customRules?: RenderRule<H, T, F>[];
 };
-export declare function StructuredText<R extends StructuredTextGraphQlResponseRecord>({ data, renderInlineRecord, renderLinkToRecord, renderBlock, renderText, renderNode, renderFragment, customMarkRules, customRules, customNodeRules, metaTransformer, }: StructuredTextPropTypes<R>): ReactElement | null;
+export declare function StructuredText<R1 extends StructuredTextGraphQlResponseRecord, R2 extends StructuredTextGraphQlResponseRecord>({ data, renderInlineRecord, renderLinkToRecord, renderBlock, renderText, renderNode, renderFragment, customMarkRules, customRules, customNodeRules, metaTransformer, }: StructuredTextPropTypes<R1, R2>): ReactElement | null;
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by stefanoverna** [comment]
Thanks Sung! Would you mind submitting a proper PR please?

—
Stefano Verna
Founder and CEO @ DatoCMS

--- original message ---
On February 6, 2022 at 4:33 PM GMT+1 ***@***.*** wrote:



Hi! 👋

Firstly, thanks for your work on this project! 🙂

Today I used patch-package to patch ***@***.*** for the project I'm working on.

This issue is a follow-up from this forum post.

Here is the diff that solved my problem:

diff --git a/node_modules/react-datocms/dist/types/StructuredText/index.d.ts b/node_modules/react-datocms/dist/types/StructuredText/index.d.ts
index 5e55966..1ea35fd 100644
--- a/node_modules/react-datocms/dist/types/StructuredText/index.d.ts
+++ b/node_modules/react-datocms/dist/types/StructuredText/index.d.ts
@@ -25,19 +25,19 @@ declare type RenderRecordLinkContext<R extends StructuredTextGraphQlResponseReco
declare type RenderBlockContext<R extends StructuredTextGraphQlResponseRecord> = {
record: R;
};
-export declare type StructuredTextPropTypes<R extends StructuredTextGraphQlResponseRecord> = {
+export declare type StructuredTextPropTypes<R1 extends StructuredTextGraphQlResponseRecord, R2 extends StructuredTextGraphQlResponseRecord> = {
/** The actual field value you get from DatoCMS **/
- data: StructuredTextGraphQlResponse<R> | StructuredTextDocument | Node | null | undefined;
+ data: StructuredTextGraphQlResponse<R1, R2> | StructuredTextDocument | Node | null | undefined;
/** A set of additional rules to convert nodes to JSX **/
customNodeRules?: RenderRule<H, T, F>[];
/** A set of additional rules to convert marks to JSX **/
customMarkRules?: RenderMarkRule<H, T, F>[];
/** Fuction that converts an 'inlineItem' node into React **/
- renderInlineRecord?: (context: RenderInlineRecordContext<R>) => ReactElement | null;
+ renderInlineRecord?: (context: RenderInlineRecordContext<R2>) => ReactElement | null;
/** Fuction that converts an 'itemLink' node into React **/
- renderLinkToRecord?: (context: RenderRecordLinkContext<R>) => ReactElement | null;
+ renderLinkToRecord?: (context: RenderRecordLinkContext<R2>) => ReactElement | null;
/** Fuction that converts a 'block' node into React **/
- renderBlock?: (context: RenderBlockContext<R>) => ReactElement | null;
+ renderBlock?: (context: RenderBlockContext<R1>) => ReactElement | null;
/** Function that converts 'link' and 'itemLink' `meta` into HTML props */
metaTransformer?: TransformMetaFn;
/** Fuction that converts a simple string text into React **/
@@ -49,4 +49,4 @@ export declare type StructuredTextPropTypes<R extends StructuredTextGraphQlRespo
/** @deprecated use customNodeRules **/
customRules?: RenderRule<H, T, F>[];
};
-export declare function StructuredText<R extends StructuredTextGraphQlResponseRecord>({ data, renderInlineRecord, renderLinkToRecord, renderBlock, renderText, renderNode, renderFragment, customMarkRules, customRules, customNodeRules, metaTransformer, }: StructuredTextPropTypes<R>): ReactElement | null;
+export declare function StructuredText<R1 extends StructuredTextGraphQlResponseRecord, R2 extends StructuredTextGraphQlResponseRecord>({ data, renderInlineRecord, renderLinkToRecord, renderBlock, renderText, renderNode, renderFragment, customMarkRules, customRules, customNodeRules, metaTransformer, }: StructuredTextPropTypes<R1, R2>): ReactElement | null;


This issue body was partially generated by patch-package.

—

Reply to this email directly, view it on GitHub, or unsubscribe.

Triage notifications on the go with GitHub Mobile for iOS or Android. 

You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>
--- end of original message ---

**Comment by BasixKOR** [comment]
Yes that would be a better option. It seems like the type is actually from the `datocms-structured-text-utils` so I'll make a PR there.

**Comment by stefanoverna** [comment]
Released as 3.0.7, thanks @BasixKOR!
